File: VB/E4915/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace E4915.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
            Return View()
        End Function

        Public Function Export() As ActionResult
            Dim model = Product.GetProducts()
            Return GridViewExtension.ExportToPdf(ExportHelper.GetGridViewSettings(True), model)
        End Function

        <ValidateInput(False)>
        Public Function GridViewPartial() As ActionResult
            Dim model = Product.GetProducts()
            Return PartialView("_GridViewPartial", model)
        End Function

        Public NotInheritable Class ExportHelper

            Shared Public Function GetGridViewSettings(ByVal isExport As Boolean) As GridViewSettings
                Dim settings As GridViewSettings = New GridViewSettings()
                settings.Name = "GridView"
                settings.CallbackRouteValues = New With {.Controller = "Home", .Action = "GridViewPartial"}
                settings.KeyFieldName = "ProductID"
                settings.Columns.Add("ProductID")
                settings.Columns.Add("ProductName")
                If isExport Then
                    'Columns only to export
                    settings.Columns.Add("UnitsInStock")
                    settings.Columns.Add("UnitsOnOrder")
                Else
                    'Columns only to display on the web page
                    settings.Columns.Add("UnitPrice")
                End If

                Return settings
            End Function
        End Class
    End Class
End Namespace

File: CS/E4915/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace E4915.Controllers
{
public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";
            return View();
        }

        public ActionResult Export() {
            var model = Product.GetProducts();
            return GridViewExtension.ExportToPdf(ExportHelper.GetGridViewSettings(true), model);
        }

        [ValidateInput(false)]
        public ActionResult GridViewPartial() {
            var model = Product.GetProducts();
            return PartialView("_GridViewPartial", model);
        }

        public static class ExportHelper {
            static public GridViewSettings GetGridViewSettings(bool isExport) {
                GridViewSettings settings = new GridViewSettings();
                settings.Name = "GridView";
                settings.CallbackRouteValues = new {
                    Controller = "Home",
                    Action = "GridViewPartial"
                };

                settings.KeyFieldName = "ProductID";
                settings.Columns.Add("ProductID");
                settings.Columns.Add("ProductName");
                if (isExport) {
                    //Columns only to export
                    settings.Columns.Add("UnitsInStock");
                    settings.Columns.Add("UnitsOnOrder");
                }
                else {
                    //Columns only to display on the web page
                    settings.Columns.Add("UnitPrice");
                }
                return settings;
            }
        }
    }
}
